pyro/infer/reparam/transform.py

```diff
@@ -0,0 +1,45 @@
+import pyro
+import pyro.distributions as dist
+
+
+class TransformReparam:
+    """
+    Arbitrary transform reparameterizer.
+
+    This can be used to reparameterize wrt an arbitrary bijective
+    :class:`~torch.distributions.transforms.Transform` object, and requires:
+
+    - for sample sites, the forward ``.__call__()`` method and the
+      ``.log_abs_det_jacobian()`` transform to be defined, as in [1]
+    - for observation sites, requires the backward ``.inv.__call__()`` method
+      and the ``.log_abs_det_jacobian()``.
+
+    [1] Hoffman, M. et al. (2019)
+        "NeuTra-lizing Bad Geometry in Hamiltonian Monte Carlo Using Neural Transport"
+        https://arxiv.org/abs/1903.03704
+
+    :param ~pyro.distributions.TorchDistribution base_dist: A base
+        distribution for the auxiliary latent variable.
+    :param ~torch.distributions.transforms.Transform transform: A bijective
+        transform defining forward and log abs det jacobian methods.
+    """
+    def __init__(self, base_dist, transform):
```

Comment:
I am not sure how useful this TransformReparam is, but I would prefer a TransformReparam which allows me to reparameterize the following site:
```
x = pyro.sample("x", TransformedDistribution(dist.Normal(0, 1), Affine(loc, scale)))
```
I expect the usage should be similar to `LocScaleReparam`.

Reply:
Good idea! I have updated as you suggest.